presto-main/src/main/java/io/prestosql/sql/planner/iterative/rule/MergeLimitWithDistinct.java

```diff
@@ -22,29 +22,19 @@
 import io.prestosql.sql.planner.plan.LimitNode;
 
 import static io.prestosql.matching.Capture.newCapture;
+import static io.prestosql.sql.planner.plan.Patterns.Aggregation.isDistinct;
 import static io.prestosql.sql.planner.plan.Patterns.aggregation;
 import static io.prestosql.sql.planner.plan.Patterns.limit;
 import static io.prestosql.sql.planner.plan.Patterns.source;
+import static java.lang.Boolean.TRUE;
 
 public class MergeLimitWithDistinct
         implements Rule<LimitNode>
 {
     private static final Capture<AggregationNode> CHILD = newCapture();
 
     private static final Pattern<LimitNode> PATTERN = limit()
-            .with(source().matching(aggregation().capturedAs(CHILD)
-                    .matching(MergeLimitWithDistinct::isDistinct)));
-
-    /**
-     * Whether this node corresponds to a DISTINCT operation in SQL
-     */
-    private static boolean isDistinct(AggregationNode node)
-    {
-        return node.getAggregations().isEmpty() &&
-                !node.getGroupingKeys().isEmpty() &&
-                node.getOutputSymbols().size() == node.getGroupingKeys().size() &&
-                node.getOutputSymbols().containsAll(node.getGroupingKeys());
-    }
+            .with(source().matching(aggregation().capturedAs(CHILD).with(isDistinct().equalTo(TRUE))));
```

Comment:
```
 .with(source().matching(aggregation().capturedAs(CHILD).with(AggregationNode::isDistinct);
```
?

Reply:
Moved to the above model instead of using `Patterns`